src/storage/btree_load.c

```diff
@@ -4852,17 +4897,112 @@ online_index_builder (THREAD_ENTRY * thread_p, BTID_INT * btid_int, HFID * hfids
 	}
 
       /* Dispatch the insert operation */
-      ret = btree_online_index_dispatcher (thread_p, btid_int->sys_btid, p_dbvalue, &class_oids[cur_class], &cur_oid,
-					   unique_pk, BTREE_OP_ONLINE_INDEX_IB_INSERT, NULL);
-      /* Clear the index key. */
-      pr_clear_value (p_dbvalue);
+      index_builder_loader_task *load_task = new index_builder_loader_task (btid_int->sys_btid, &class_oids[cur_class],
+									    &cur_oid, unique_pk, load_context);
 
-      if (ret != NO_ERROR)
+      load_task->set_key (p_dbvalue);
+
+      cubthread::get_manager ()->push_task (ib_workpool, load_task);
+
+      /* Increment tasks started. */
+      tasks_started++;
+
+      /* Check for possible errors. */
+      if (load_context.m_error_code != NO_ERROR)
```

Comment:
Use m_has_error to check error status of loader. It is more reliable than error code.